cli/cmd/version.go

```diff
@@ -18,43 +18,94 @@ package cmd
 
 import (
 	"fmt"
+	"os"
 	"strings"
 
 	"github.com/spf13/cobra"
 
 	"github.com/docker/compose-cli/cli/cmd/mobyflags"
 	"github.com/docker/compose-cli/cli/mobycli"
+	"github.com/docker/compose-cli/formatter"
 )
 
+const formatOpt = "format"
+
 // VersionCommand command to display version
 func VersionCommand(version string) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "version",
 		Short: "Show the Docker version information",
 		Args:  cobra.MaximumNArgs(0),
-		RunE: func(cmd *cobra.Command, _ []string) error {
-			return runVersion(cmd, version)
+		Run: func(cmd *cobra.Command, _ []string) {
+			runVersion(cmd, version)
 		},
 	}
 	// define flags for backward compatibility with com.docker.cli
 	flags := cmd.Flags()
-	flags.StringP("format", "f", "", "Format the output using the given Go template")
+	flags.StringP(formatOpt, "f", "", "Format the output using the given Go template")
+	// flags.String(&opts.format, "format", "", "Format the output. Values: [pretty | json | go template]. (Default: pretty)")
 	flags.String("kubeconfig", "", "Kubernetes config file")
 	mobyflags.AddMobyFlagsForRetrocompatibility(flags)
 
 	return cmd
 }
 
-func runVersion(cmd *cobra.Command, version string) error {
+func runVersion(cmd *cobra.Command, version string) {
+	var versionString string
+	format := strings.TrimSpace(cmd.Flag(formatOpt).Value.String())
 	displayedVersion := strings.TrimPrefix(version, "v")
-	versionResult, _ := mobycli.ExecSilent(cmd.Context())
+	// Replace is preferred in this case to keep the order.
+	switch format {
+	case formatter.PRETTY, "":
+		versionString = strings.Replace(getOutFromMoby(cmd, fixedPrettyArgs(os.Args[1:])...),
+			"\n Version:", "\n Cloud integration:  "+displayedVersion+"\n Version:", 1)
+	case formatter.JSON, "{{json .}}", "{{json . }}", "{{ json .}}", "{{ json . }}": // Try to catch full JSON formats
```

Comment:
easier if you replace space first ? (`replace(" ", "")`)

Reply:
and toLowerCase

Reply:
Done